op-challenger/game/scheduler/coordinator.go

```diff
@@ -49,18 +52,43 @@ func (c *coordinator) schedule(ctx context.Context, games []common.Address) erro
 		}
 	}
 
+	var gamesInProgress int
+	var gamesChallengerWon int
+	var gamesDefenderWon int
 	var errs []error
+	var jobs []job
 	// Next collect all the jobs to schedule and ensure all games are recorded in the states map.
 	// Otherwise, results may start being processed before all games are recorded, resulting in existing
 	// data directories potentially being deleted for games that are required.
-	var jobs []job
 	for _, addr := range games {
+		var status types.GameStatus
+		statusSet := false
 		if j, err := c.createJob(addr); err != nil {
 			errs = append(errs, err)
 		} else if j != nil {
 			jobs = append(jobs, *j)
+			status = j.status
+			statusSet = true
+		} else {
+			// Read from the game state
+			state, ok := c.states[addr]
+			if ok {
+				status = state.status
+				statusSet = true
+			}
+		}
+		if statusSet {
```

Comment:
If a game is executing a job it _must_ have a `gameState` in the map so I don't think we need these two separate paths to get the data.  If we're going to get the status from the map, just do that all the time.

Also since the status is expected to be present, we should log an error if its not there rather than silently ignoring.